plugins/package-managers/pub/src/main/kotlin/Pub.kt

```diff
@@ -496,12 +489,12 @@ class Pub(
         return ProjectAnalyzerResult(Project.EMPTY, emptySet(), listOf(issue))
     }
 
-    private fun parseProject(definitionFile: File, pubspec: JsonNode, scopes: Set<Scope>): Project {
+    private fun parseProject(definitionFile: File, pubspec: Pubspec?, scopes: Set<Scope>): Project {
```

Comment:
It seems that the only caller passes in a non-null `Pubspec`. So can the code be simplified accordingly?